pkg/apis/extensions/types.go

```diff
@@ -40,6 +40,14 @@ const (
 	// allowed safe and unsafe sysctls in a pod spec. It's a comma-separated list of plain sysctl
 	// names or sysctl patterns (which end in *). The string "*" matches all sysctls.
 	SysctlsPodSecurityPolicyAnnotationKey string = "security.alpha.kubernetes.io/sysctls"
+
+	// IngressPreSharedCertAnnotationKey represents the specific pre-shared SSL
+	// certicate for the Ingress controller to use. The controller *does not*
+	// manage this certificate, it is the users responsibility to create/delete it.
+	// In GCE, the Ingress controller assigns the SSL certificate with this name
+	// to the target proxies of the Ingress.
+	// TODO: In Nginx.
+	IngressPreSharedCertAnnotationKey = "kubernetes.io/ingress.pre-shared-cert"
```

Comment:
I wouldn't mind making this GCE specific for now. I think there was the intention to make this generic but I don't have the expertise in the Nginx controller to make changes there.

Totally open to changing the key when decided upon.

Is there a reason for alpha @smarterclayton? The [guidelines](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#label-selector-and-annotation-conventions) do not make it very clear when annotations are promoted alpha -> beta -> stable -> spec.